File: src/Cardiologista.ts
import { Especialista } from "./Especialista";
import { DadosSintomas } from "./DadosSintomas";

export class Cardiologista extends Especialista {
  protected podeDiagnosticar(dados: DadosSintomas): boolean {
    return dados.sintomas.some(s =>
      ["dor no peito", "pressão alta", "taquicardia"].includes(s.toLowerCase())
    );
  }

  protected diagnosticar(dados: DadosSintomas): string {
    return "Diagnóstico pelo Cardiologista: possível problema cardíaco.";
  }
}

File: SistemaDiagnostico/Patterns/src/Especialistas/Cardiologista.java
package Especialistas;

import Diagnostico.DadosSintomas;
import java.util.Arrays;

public class Cardiologista extends Especialista {
    @Override
    protected boolean podeDiagnosticar(DadosSintomas dados) {
        return dados.getSintomas().stream()
            .map(String::toLowerCase)
            .anyMatch(s -> Arrays.asList("dor no peito", "pressão alta", "taquicardia").contains(s));
    }

    @Override
    protected String diagnosticar(DadosSintomas dados) {
        return "Diagnóstico pelo Cardiologista: possível problema cardíaco.";
    }
}
